tests/test_dataclasses.py

```diff
@@ -131,20 +133,6 @@ def double_a(cls, v):
     assert d.a == 6
 
 
-def test_validate_assignment_extra():
-    @pydantic.dataclasses.dataclass(config=dict(validate_assignment=True), frozen=False)
-    class MyDataclass:
-        a: int
-
-    d = MyDataclass(1)
-    assert d.a == 1
-
-    d.extra_field = 1.23
-    assert d.extra_field == 1.23
-    d.extra_field = 'bye'
-    assert d.extra_field == 'bye'
```

Comment:
I think this was removed because you can't set missing fields with validate_assignment=True, but I would suggest we add a test of the new behavior:
```python
def test_validate_assignment_extra_missing_fields():
    @pydantic.dataclasses.dataclass(config=dict(validate_assignment=True), frozen=False)
    class MyDataclass:
        a: int

    d = MyDataclass(1)
    assert d.a == 1

    with pytest.raises(ValidationError) as exc_info:
        d.extra_field = 1.23
    assert exc_info.value.errors() == [
        {
            'ctx': {'attribute': 'extra_field'},
            'input': 1.23,
            'loc': ('extra_field',),
            'msg': "Object has no attribute 'extra_field'",
            'type': 'no_such_attribute',
        }
    ]
```

Reply:
Whoops, no, those are only for BaseModel not dataclass

Reply:
That test only works because `extra` in `ConfigDict` is ignored and `forbid` is used as the default. So we'd have to rewrite that test once we get `extra` (and `frozen`) working for dataclasses. So can we table it until then?

Reply:
In other words, I think the tests we want are:

```python
@pytest.mark.parametrize('config', [
    ConfigDict(validate_assignment=True),
    ConfigDict(validate_assignment=True, extra=Extra.ignore)
])
def test_validate_assignment_extra_ignore(config: ConfigDict):
    @pydantic.dataclasses.dataclass(config=config)
    class MyDataclass:
        a: int

    d = MyDataclass(1)
    assert d.a == 1

    d.extra_field = 1.23
    assert not hasattr(d, 'extra_field')


def test_validate_assignment_extra_forbid():
    @pydantic.dataclasses.dataclass(config=ConfigDict(validate_assignment=True, extra=Extra.forbid))
    class MyDataclass:
        a: int

    d = MyDataclass(1)
    assert d.a == 1

    with pytest.raises(ValidationError) as exc_info:
        d.extra_field = 1.23
    assert exc_info.value.errors() == [
        {
            'ctx': {'attribute': 'extra_field'},
            'input': 1.23,
            'loc': ('extra_field',),
            'msg': "Object has no attribute 'extra_field'",
            'type': 'no_such_attribute',
        }
    ]
```

Reply:
> So can we table it until then?

I just was hesitant to delete a test as I worry it would be forgotten. If you don't think that's a concern I'm fine tabling.

Reply:
After discussion offline, I was wrong. I added a ~ similar test to the one you suggested 😄